test/src/test/java/aima/test/unit/search/basic/BasicPriorityFrontierQueueTest.java

```diff
@@ -0,0 +1,131 @@
+package aima.test.unit.search.basic;
+
+import aima.core.search.basic.support.BasicFrontierQueue;
+import aima.core.search.basic.support.BasicNodeFactory;
+import aima.core.search.basic.support.BasicPriorityFrontierQueue;
+import aima.core.search.api.Node;
+import aima.core.search.api.NodeFactory;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.PriorityQueue;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BasicPriorityFrontierQueueTest<A, S> {
+	
+	private static NodeFactory<Integer, String> nodeFactory;
+	private static int NUMBER_OF_NODES;
+	private static int NUMBER_OF_STATES;
+	private static Node[] testNodes;
+	private static Node[] fastPQResult;
+	private static Node[] normalPQResult;
+	private static Comparator<Node<Integer, String>> comparator;
+	
+	@BeforeClass
+	public static void prepare() {
+		
+		
+		nodeFactory = new BasicNodeFactory<Integer, String>();
+		
+		//Custom comparator to avoid duplicates in a TreeSet
+		//Checks on pathCost first, followed by state (only for a clearer output), and finally followed by object equality
+		
+		comparator = (Node<Integer, String> node1, Node<Integer, String> node2)-> 
+					 ((node1.pathCost() == node2.pathCost()) ? ((node1.state().equals(node2.state())) ? (node1.equals(node2) ? 0 : 1) : (node1.state().compareTo(node2.state()))) 
+					 : (int)Math.signum((node1.pathCost() - node2.pathCost())));
+		
+	    // Small Test Cases for fastPQ

```

Comment:
I tried using a NodeContainer object, but there were many issues with using it.

I've used a rather simpler approach to fix this, and the only issue remains with implementing the iterator() function, but since its overriding a Collections method, I've left it unimplemented as of now. 

Changes are summarized as below:

1) The comparator issue is solved rather easily by just using 
comparator.thenComparing((n1,n2) -> Objects.equals(n1, n2) ? 0 : 1); 
over the comparator already received in the constructor.

Now our TreeSet only contains unique nodes. But we maintain counts of these unique nodes in a separate HashMap, to get an effect similar to a MultiTreeSet, hence meeting our requirements.